Source/drlg_l4.cpp

```diff
@@ -1445,8 +1445,9 @@ void uShape()
 				hallok[j] = FALSE;
 			}
 			if (dung[i][j] == 1) {
-				// BUGFIX: check that i + 1 < 20 and j + 1 < 20
-				if (dung[i][j + 1] == 1 && dung[i + 1][j + 1] == 0) {
+				// BUGFIX: check that i + 1 < 20 and j + 1 < 20 (fixed)
+				if (i + 1 < 20 && j + 1 < 20 &&

```

Comment:
nit for the future: `20` should probably be a constant like in the other `drlg` files

Reply:
I guess it could be `DMAXX / 2`? Any way that should be done upstream :)